pyxem/signals/beam_shift.py

```diff
@@ -62,42 +75,507 @@ def make_linear_plane(self, mask=None):
+            fx = np.fft.fftshift(np.fft.fft2(dx))
+            fy = np.fft.fftshift(np.fft.fft2(dy))
+            # weights in x,y directins, currently hardcoded, but easy to extend if required
+            wx, wy = 0.5, 0.5
+
+            numerator = -1j * (wx * kx_grid * fx + wy * ky_grid * fy)
+            denominator = wx * kx_grid**2 + wy * ky_grid**2
+
+        # handle the division by zero in the central pixel
+        # set the undefined/infinity pixel to 0
+        with np.errstate(divide="ignore", invalid="ignore"):
+            res = numerator / denominator
+        res = np.nan_to_num(res, nan=0, posinf=0, neginf=0)
+
+        retrieved = np.fft.ifft2(np.fft.ifftshift(res)).real
+
+        # get 1/4 of the result if mirroring
+        if mirroring:
+            M, N = retrieved.shape
+            retrieved = retrieved[: M // 2, : N // 2]
+
+        signal = self._deepcopy_with_new_data(retrieved)
+        signal._remove_axis(-1)
+        signal = signal.T
+        signal.metadata.General.title = "Phase retrieval of {0}".format(
+            self.metadata.General.title
+        )
+        return signal
+
+    def get_phase_signal(self, rotation=None):
+        """Get beam shift phase image visualized using continuous color scale.
+
+        Converts the x and y beam shifts into an RGB array, showing the
+        direction of the beam shifts.
+
+        Useful for visualizing magnetic domain structures.
+
+        Parameters
+        ----------
+        rotation : float, optional
+            In degrees. Useful for correcting the mismatch between
+            scan direction and diffraction pattern rotation.
+        autolim : bool, default True
+        autolim_sigma : float, default 4
+
+        Returns
+        -------
+        phase_signal : HyperSpy 2D RGB signal
+
+        Examples
+        --------
+        >>> s = pxm.data.dummy_data.get_simple_beam_shift_signal()
+        >>> s_color = s.get_phase_signal(rotation=20)
+        >>> s_color.plot()
+
+        See Also
+        --------
+        get_color_signal : Signal showing both phase and magnitude
+        get_magnitude_signal : Signal showing the magnitude
+
+        """
+
+        if self.axes_manager.navigation_dimension != 2:
+            raise ValueError("get_phase_signal only works with 2 navigation dimensions")
+        # Rotate the phase by -30 degrees in the color "wheel", to get better
+        # visualization in the vertical and horizontal direction.
+        if rotation is None:
+            rotation = -30
+        else:
+            rotation = rotation - 30
+
+        phase = np.arctan2(self.isig[0].data, self.isig[1].data) % (2 * np.pi)
+        rgb_array = bst._get_rgb_phase_array(phase=phase, rotation=rotation)
+        rgb_array_16bit = rgb_array * (2**16 - 1)
+        s_rgb = self._deepcopy_with_new_data(rgb_array_16bit)
+        s_rgb.change_dtype("uint16")
+        s_rgb.change_dtype("rgb16")
+        return s_rgb
+
+    def get_color_signal(
```

Comment:
Would it be a good idea to rename this function? As it can easily be confused with `get_phase_signal` since it's also very colourful. Maybe something like `get_magnitude_phase_signal`. Easier to search for too.